python/argo_client/connection.py

```diff
@@ -472,6 +473,11 @@ def send_notification(self, method: str, params: dict) -> None:
                'params': params}
         msg_string = json.dumps(msg)
         self.process.send_one_message(msg_string, expecting_response = False)
+        # Since notifications require no response and can interrupt commands/queries,
+        # they can sometimes interrupt subsequent requests. We have tried to guarantee
+        # in-order servicing on the server. This short pause is an additional small
+        # safety measure to keep things running smoothly.
+        time.sleep(0.1)
```

Comment:
This is somewhat unfortunate, and seems like it's fragile in the long run. I'm happy to include it for now, to get things working, but I'd hope that we can figure out a way to not need it in the long run.

Reply:
I just went back and tested this a final time, making sure the cryptol server was in fact using the _new_ argo updates in this PR, and the explicit pause is no longer necessary to avoid the unknown state id errors even in the pathological looping case. I'll confirm again tomorrow when I do the cryptol bump, but I'm pretty sure we're good-to-go now. (Last time I must have updated the client but not build and _reinstalled_ the server, doh.)